contracts/extensions/TradeExtension.sol

```diff
@@ -0,0 +1,147 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+
+import { BaseGlobalExtension } from "../lib/BaseGlobalExtension.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { ITradeModule } from "../interfaces/ITradeModule.sol";
+
+/**
+ * @title TradeExtension
+ * @author Set Protocol
+ *
+ * Smart contract global extension which provides DelegatedManager privileged operator(s) the ability 
+ * to trade on a DEX and the owner the ability to restrict operator(s) permissions with an asset whitelist.
+ */
+contract TradeExtension is BaseGlobalExtension {
+
+    /* ============ Events ============ */
+
+    event ExtensionInitialized(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    /* ============ State Variables ============ */
+
+    // Instance of TradeModule
+    ITradeModule public immutable tradeModule;
+
+    // Mapping from Set Token to DelegatedManager 
+    mapping(ISetToken => IDelegatedManager) public setManagers;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        ITradeModule _tradeModule
+    )
+        public
+    {
+        tradeModule = _tradeModule;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ONLY OWNER: Initializes TradeExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) public {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        setManagers[_delegatedManager.setToken()] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        ExtensionInitialized(address(_delegatedManager.setToken()), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes TradeExtension to the DelegatedManager and TradeModule to the SetToken
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeModuleAndExtension(IDelegatedManager _delegatedManager) external {
+        require(_delegatedManager.setToken().isPendingModule(address(tradeModule)), "TradeModule must be pending");
+
+        initializeExtension(_delegatedManager);
+
+        bytes memory callData = abi.encodeWithSignature(
+            "initialize(ISetToken)", 
+            _delegatedManager.setToken()
+        );
+        address(_delegatedManager).call(callData);
```

Comment:
is this the right way to initialize the tradeModule from the input _delegatedManager

Reply:
we have a function that does this for you in the `BaseGlobalExtension`. Also what's the interface you're trying to call on the manager? This is the initialize interface: `function initializeExtension() external;`

Reply:
you should be calling the manager by using: `invokeManager`. Any interactions with the manager need to go through there. Not a raw call